src/app/homeSignIn/homeSignIn.component.js

```diff
@@ -0,0 +1,62 @@
+import angular from 'angular';
+import includes from 'lodash/includes';
+
+import commonModule from 'common/common.module';
+import loadingOverlay from 'common/components/loadingOverlay/loadingOverlay.component';
+import sessionService from 'common/services/session/session.service';
+import sessionModalService from 'common/services/session/sessionModal.service';
+
+import template from './homeSignIn.tpl';
+
+let componentName = 'homeSignIn';
+
+class HomeSignInController {
+
+  /* @ngInject */
+  constructor(sessionService, sessionModalService) {
+    this.sessionService = sessionService;
+    this.sessionModalService = sessionModalService;
+  }
+
+  $onInit() {
+    this.isSigningIn = false;
+    this.showSignInForm = true;
+
+    if ( includes( ['IDENTIFIED', 'REGISTERED'], this.sessionService.getRole() ) ) {
+      this.showSignInForm = false;
+      this.username = this.sessionService.session.email;
+    }
+  }
+
+  signIn() {
+    this.isSigningIn = true;
+    delete this.errorMessage;
+
+    this.sessionService
+      .signIn( this.username, this.password )
+      .subscribe( () => {
+        this.isSigningIn = false;
+        this.showSignInForm = false;
+      }, ( error ) => {
+        this.isSigningIn = false;
+        this.errorMessage = error.data.error;
+      } );
+  }
+
+  signUp() {
+    this.sessionModalService.signUp();
```

Comment:
Can you test this too? I think it would just be a single spy to check if it has been called.